### ISSUE TITLE
🪣 XP Action Event Projections

### ISSUE THREAD

**Issue by dillchen** [hicommonwealth/commonwealth#9343]
### Description

These projections will track and log user engagement across various activities, some of which are not on our platform, contributing to the XP system and broader user engagement metrics.

### Project Owner

@dillchen

### User Stories / Acceptance Criteria

Each projection should track the event and log the corresponding points or progress towards quests.
Ensure projections update user profiles and XP totals as specified by the associated schema.

Create projections for each of the following actions. Not all are necessary or launch blocking
- [ ] **Launch Blocking**: Link Wallet
- [ ] **Launch Blocking**: Link Farcaster
- [ ] **Launch Blocking**: Link SSO
- [ ] **Launch Blocking**: Share Initial Post on Twitter
- [ ] **Launch Blocking**: Follow on Telegram
- [ ] **Launch Blocking**: Follow on Farcaster
- [ ] **Launch Blocking**: Follow on Twitter
- [ ] **Launch Blocking**: User comes back for first 7 days to claim daily XP
- [ ] User mentions event - This got removed from client in https://github.com/hicommonwealth/commonwealth/issues/10825, need to add back when event projection is complete in platform.

Later:
- [ ] GM Stream Completed
- [ ] Tapping -> Earning $NEWCOIN
- [ ] Create Coin with Purchase = $NEWCOIN + Clubs
- [ ] Creates an Upvoted Post
- [ ] Daily Quest (Pack of 3)

### Platform Tasks
- [x] #9399
- [x] #10078
- [ ] #10080
- [ ] #10081
- [x] #10082
- [x] #10145
- [x] #10188

### Additional Context

Here is the full list of events, not all of these are on [platform](https://docs.google.com/spreadsheets/d/1LL8jGRwCaqrG_bpYW5WW3CfjN-vVEYkIbvY91Bwg4-g/edit?gid=2005659128#gid=2005659128)

**Comment by dillchen** [comment]
from @Rotorsoft there are a set of projections that are internal to our platform so are unblocked, include Dillon in the convo on the ones that are unblocked.

We may split these into different batches
